lib/redirect.js

```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+const { full_url_for } = require('hexo-util');
+
+module.exports = function(data) {
+  const { alias, aliases, redirect } = data;
+  if (redirect && !alias && !aliases) {
+    const target = full_url_for.call(this, redirect);
+    data.layout = '';
+    data.content = '<!DOCTYPE html>'
+    + '<html>'
+    + '<head>'
+      + '<meta charset="utf-8">'
+      + '<title>Redirecting...</title>'
+      + '<link rel="canonical" href="' + target + '">'
```

Comment:
`target` must be absolute url (`http://example.com/red/green/`) not relative (`/red/green`),
https://support.google.com/webmasters/answer/139066#rel-canonical-link-method